### ISSUE TITLE
Unsupported argument (not_resources) with aws 3.69.0 provider

### ISSUE THREAD

**Issue by zeusal** [lgallard/terraform-aws-backup#58]
Hi, 

I am using this module in my terraforming code but after looking at the module documentation, the module code and the official terraforming documentation of the AWS resource it generates, I see some inconsistencies.

In the module documentation it says versions greater than or equal to 3.20.0 provider ([aws ](https://registry.terraform.io/providers/hashicorp/aws)(hashicorp/aws) >= 3.20.0), I am using 3.69.0 and in the official terraforming documentation of the version I am using, it does not contemplate the not_resources argument (https://registry.terraform.io/providers/hashicorp/aws/3.69.0/docs/resources/backup_selection) so it generates an error, I had to use the 0.12.1 version of the module to avoid this error.

![image](https://user-images.githubusercontent.com/33123154/163172459-dae0ebf8-6d37-44aa-8806-39474202ca47.png)

This occurs with module versions higher than 0.12.1..

Regards!

Related to #50

**Comment by lgallard** [comment]
@zeusal thanks for reporting this. At first sight I would have to use the mixed provider definition to be compatible with version 0.12.x and 0.13.x as explained [here](https://www.terraform.io/language/providers/requirements#v0-12-compatible-provider-requirements), and also check the compatibility among versions.

I'll be working on this, and depending on my findings I can create another release or comment them here.

**Comment by lgallard** [comment]
@zeusal version [0.15.0 ](https://github.com/lgallard/terraform-aws-backup/releases/tag/0.15.0)sets the right constraints for Terraform version and AWS providers to use with this module. 

**TL;DR**
---
Min Terraform version: `0.12.31`
Min AWS provider: `3.72.0`
